dace/codegen/cppunparse.py

```diff
@@ -811,43 +811,16 @@ def _Num(self, t, infer_type=False):
         # double-complex (128-bit) otherwise
         dtype = self.dtype or 'dace::complex128'
 
-        if six.PY3:
-            if repr_n.endswith("j"):
-                self.write("%s(0, %s)" %
-                           (dtype, repr_n.replace("inf", INFSTR)[:-1]))
-            else:
-                self.write(repr_n.replace("inf", INFSTR), infer_type)
-                # If the number has a type, use it
-                if isinstance(t.n, np.uint):
-                    return dace.dtypes.typeclass(
-                        np.uint32) if infer_type else None
-                elif isinstance(t.n, np.int):
-                    return dace.dtypes.typeclass(
-                        np.int32) if infer_type else None
-                elif isinstance(t.n, np.float):
-                    return dace.dtypes.typeclass(
-                        np.float32) if infer_type else None
-                elif isinstance(t.n, np.float64):
-                    return dace.dtypes.typeclass(
-                        np.float64) if infer_type else None
-                elif infer_type:
-                    raise TypeError('Unable to convert number')
+        if repr_n.endswith("j"):
+            self.write("%s(0, %s)" %
+                       (dtype, repr_n.replace("inf", INFSTR)[:-1]))
         else:
-            # Parenthesize negative numbers, to avoid turning (-1)**2 into -1**2.
-            if repr_n.startswith("-"):
-                self.write("(")
-            if "inf" in repr_n and repr_n.endswith("*j"):
-                repr_n = repr_n.replace("*j", "j")
-
-            if repr_n.endswith("j"):
-                self.write("%s(0, %s)" %
-                           (dtype, repr_n.replace("inf", INFSTR)[:-1]))
+            self.write(repr_n.replace("inf", INFSTR), infer_type)
+            # If the number has a type, use it
+            if infer_type:
+                return dace.dtypes.typeclass(type(t.n))
             else:
-                # Substitute overflowing decimal literal for AST infinities.
-                self.write(repr_n.replace("inf", INFSTR))
-
-            if repr_n.startswith("-"):
-                self.write(")")
```

Comment:
what happened to this?

Reply:
It's above, just diff makes it look like it was deleted and re-added